src/test/java/de/tum/in/www1/artemis/ComplaintResponseIntegrationTest.java

```diff
@@ -0,0 +1,551 @@
+
+        request.put("/api/complaint-responses/complaint/" + complaint.getId() + "/resolve", complaintResponseToBeUsedInResolve, HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void resolveComplaint_LockActiveBUTInstructor_shouldResolve() throws Exception {
+        ComplaintResponse initialLock = createLockOnComplaint("tutor2", false);
+        assertThat(initialLock.isCurrentlyLocked()).isTrue();
+
+        ComplaintResponse complaintResponseToBeUsedInResolve = initialLock;
+        complaintResponseToBeUsedInResolve.setResponseText("Accepted");
+        complaintResponseToBeUsedInResolve.getComplaint().setAccepted(true);
+
+        request.put("/api/complaint-responses/complaint/" + complaint.getId() + "/resolve", complaintResponseToBeUsedInResolve, HttpStatus.OK);
+        validateThatComplaintIsResolved("instructor1");
+    }
+
+    @Test
+    @WithMockUser(username = "tutor2", roles = "TA")
+    public void resolveComplaint_LockActiveBUTCreatorOfLock_shouldResolve() throws Exception {
+        ComplaintResponse initialLock = createLockOnComplaint("tutor2", false);
+        assertThat(initialLock.isCurrentlyLocked()).isTrue();
+
+        ComplaintResponse complaintResponseToBeUsedInResolve = initialLock;
+        complaintResponseToBeUsedInResolve.setResponseText("Accepted");
+        complaintResponseToBeUsedInResolve.getComplaint().setAccepted(true);
+
+        request.put("/api/complaint-responses/complaint/" + complaint.getId() + "/resolve", complaintResponseToBeUsedInResolve, HttpStatus.OK);
+        validateThatComplaintIsResolved("tutor2");
+    }
+
+    public void validateThatComplaintIsResolved(String loginOfResolver) {
+        Optional<ComplaintResponse> optionalComplaintResponse = complaintResponseRepository.findByComplaint_Id(complaint.getId());
+        assertThat(optionalComplaintResponse.isPresent()).isTrue();
+        ComplaintResponse complaintResponse = optionalComplaintResponse.get();
+        assertThat(complaintResponse.getComplaint().isAccepted()).isTrue();
+        assertThat(complaintResponse.getSubmittedTime()).isNotNull();
+        assertThat(complaintResponse.getResponseText()).isEqualTo("Accepted");
+        assertThat(complaintResponse.getReviewer().getLogin()).isEqualTo(loginOfResolver);
+    }
+
+    private ComplaintResponse createLockOnComplaint(String lockOwnerLogin, boolean runOut) {
+        ComplaintResponse complaintResponse = new ComplaintResponse();
+        complaintResponse.setComplaint(complaint);
+        User tutor = userRepository.findOneByLogin(lockOwnerLogin).get();
+        complaintResponse.setReviewer(tutor);
+        complaintResponse = complaintResponseRepository.saveAndFlush(complaintResponse);
+
+        // manually modifying the created date set by entity listener
+        if (runOut) {
+            Constants.COMPLAINT_LOCK_DURATION_IN_MINUTES = -5;
```

Comment:
Of course this will not work when we make the constant actually constant. Could we change the creationDate for that and save it or does the JPA Auditing not allow this? Otherwise, I would just set those created dates as normal attribute manually without JPA auditing if it makes many things more difficult.

Reply:
I had an implementation that changed the creationDate via a SQL statement. This worked but this method here was then decided to be better in discussion with @krusche .